xiilib/databases.py

```diff
@@ -54,16 +56,16 @@ def __init__(self, charm: ops.CharmBase, relation_name: str):
             charm: The requesting charm object.
             relation_name: The redis relation name.
         """
-        self._charm = charm
-        self.relation_name = relation_name
-        # redis charm doesn't provide database id via the relation
+        # we don't need store for Redis
```

Comment:
This is required by the Redis charm library, it doesn't have any use for our use case.

Reply:
So can we pass it the real storage even if we don't use it? Generally production code shouldn't have a mock in it since that is intended for use during testing

Reply:
`StoreState` is deprecated I think so I would prefer not using that

Reply:
We are forced to use it by the library, even if it is deprecated it is better than passing a mock

Reply:
okay, if that makes you happy